compiler/rustc_hir_typeck/src/expr.rs

```diff
@@ -2960,3 +2990,41 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         }
     }
 }
+
+fn expr_may_be_uninhabited(expr: &hir::Expr<'_>) -> bool {
```

Comment:
please add a comment here.  is this about which expressions can actually introduce an uninhabited type. E.g. arrays is not considered because in `[return;]` its the nested expression that is uninhabited :thinking: but then why do we return `false` for `ExprKind::Continue` and `ExprKind::Return`?